operator/v1/types_network.go

```diff
@@ -116,6 +116,15 @@ type NetworkSpec struct {
 	Migration *NetworkMigration `json:"migration,omitempty"`
 }
 
+// migrationMode declares valid SDN migration mode
+// +kubebuilder:validation:Enum:=Live;Offline;""
+type MigrationMode string
+
+const (
+	liveMigrationMode    MigrationMode = "Live"
+	offlineMigrationMode MigrationMode = "Offline"
+)
```

Comment:
Why would you define a new type without exporting its possible values?
Would we have the alternative to inline the validation in the field instead of defining a new type?

Reply:
Oh, that's a mistake. I shall export the values. But defining a new type is a common practice in the API repo. e.g. https://github.com/openshift/api/blob/fd420c26854b53f62bd704b9cb8952a82c281403/machine/v1/types_controlplanemachineset.go#L87

And in the CRD file, we can see the validation is added as expected.

Reply:
ok, let's keep it simple. I changed the type to `string`.

Reply:
Preferred way is to do a defined type for enumerated values. Those which do not currently do this are not compliant with current API guidelines